### PR TITLE
Minor API client changes

### PR DESCRIPTION
**Pull request by kiddinn**
This PR introduces the following API client changes:

1. Adds an `expected_size` property to the search object, so that one can find out how many entries one can expect back from a search query.
2. Added minor error handling into the `list_saved_searches` to account for malformed saved searches.

### CODE REVIEW

File: api_client/python/timesketch_api_client/search.py [hunk 1]
```diff
@@ -573,6 +574,28 @@ def description(self, description):
...
+        """Property that returns the expected size of the search query."""
+        if self._total_elastic_size:
+            return self._total_elastic_size
+
+        # To get the total size of the query we first need to send a query
+        # so to minimize the cost we reduce the size to 10 entries.
+        max_entries = self._max_entries
+        scrolling = self.scrolling
+
+        self._max_entries = 10
```
Comment (berggren) [inline api_client/python/timesketch_api_client/search.py:588]: There is a better way of doing this. You can pass `count=True` to the search() call and just get back the count of matching documents. This is much cheaper as it doesn't have to retrieve anything.

Ref: https://github.com/google/timesketch/blob/master/timesketch/lib/datastores/elastic.py#L344

File: api_client/python/timesketch_api_client/search.py [hunk 2]
```diff
@@ -573,6 +574,28 @@ def description(self, description):
         self._description = description
         self.commit()
 
+    @property
+    def expected_size(self):
```
Comment (berggren) [inline api_client/python/timesketch_api_client/search.py:578]: Why do we need this in the first place?

**Review by berggren — CHANGES_REQUESTED** [review]

File: api_client/python/timesketch_api_client/search.py (hunk 1 above)
Comment (kiddinn) [reply api_client/python/timesketch_api_client/search.py]: I was attempting to remove the need to make changes in the actual API... made the changes and am now using the `count`

File: api_client/python/timesketch_api_client/search.py (hunk 2 above)
Comment (kiddinn) [reply api_client/python/timesketch_api_client/search.py:587]: explained offline, for automation in order to be able to effectively set the max_entries size

**Review by kiddinn — COMMENTED** [review]
done

**Review by berggren — APPROVED** [review]

**Pull request merged**